cmd/divide.go

```diff
@@ -0,0 +1,272 @@
+		Args:    cobra.MinimumNArgs(1),
+		Aliases: []string{"d", "div"},
+		Example: divideExample,
+		PreRunE: divideArgumentValidator,
+		RunE:    executeDivide,
+	}
+)
+
+func init() {
+	divideCmd.Flags().Int64SliceP("users", "u", []int64{}, "Divides by desired users instead of equal divisions. Say you want your network divided into 32 users, 10 users, and 17 users: 'cidr divide <network> 32,10,17' ")
+	rootCmd.AddCommand(divideCmd)
+}
+
+// Validates the arguments for the network division operation.
+// Catches network parsing errors and invalid divisors.
+func divideArgumentValidator(cmd *cobra.Command, args []string) error {
+	// Network parsing checks.
+	network, err := core.ParseCIDR(args[0])
+	if err != nil {
+		return fmt.Errorf("%s\n", err)
+	}
+
+	maskSize, _ := network.Mask.Size()
+	if helper.IsIPv4Network(network) && (maskSize == 32) {
+		return fmt.Errorf("Cannot divide a %s -- No Space\n", network.String())
+	}
+	if maskSize >= 128 {
+		return fmt.Errorf("Cannot divide a %s -- No Space\n", network.String())
+	}
+
+	// Divisor and -u Users validation.
+	var divisor int64
+	var desiredUsers []int64
+	desiredUsersPerSubnet, err := cmd.Flags().GetInt64Slice("users")
+	if err != nil {
+		return err
+	}
+	if len(desiredUsersPerSubnet) == 0 {
+		if len(args) < 2 {
+			return fmt.Errorf("Enter at least one divisor\n")
+		}
+		div, err := strconv.ParseInt(args[1], 10, 64)
+		divisor = div
+		if err != nil || div <= 1 {
+			return fmt.Errorf("%s\n", err)
+		}
+	} else {
+		desiredUsers, err = ValidateUserSpace(network, desiredUsersPerSubnet)
+		if err != nil {
+			return fmt.Errorf("%v\n", err)
+		}
+	}
+
+	// Collects the valid arguments, so we don't need to bother with these checks later.
+	validArgs := &DivideOptions{
+		Network: network,
+		Divisor: divisor,
+		Users:   desiredUsers,
+	}
+	cmd.SetContext(context.WithValue(cmd.Context(), ArgsKey{}, validArgs))
+	return nil
+}
+
+func executeDivide(cmd *cobra.Command, args []string) error {
+	validArgsInterface := cmd.Context().Value(ArgsKey{})
+	if validArgsInterface == nil {
+		return fmt.Errorf("validArgs not in context\n")
+	}
+	validArgs, ok := validArgsInterface.(*DivideOptions)
+	if !ok {
+		return fmt.Errorf("Invalid type for validArgs: %T", validArgsInterface)
+	}
+	desiredUsers, err := cmd.Flags().GetInt64Slice("users")
+	if err != nil {
+		return err
+	}
+	if len(validArgs.Users) < 1 {
+		networks, err := DivideCidr(validArgs.Network, validArgs.Divisor)
+		if err != nil {
+			return err
+		}
+		printNetworkPartitions(networks)
+	} else {
+		networks, err := DivideCidrHosts(validArgs.Network, validArgs.Users)
+		if err != nil {
+			return err
+		}
+		printNetworkUserBreakdown(networks, desiredUsers)
+
+	}
+	return nil
+}
+
+// Divides the given network into N smaller networks.
+// Errors if division is not possible.
+func DivideCidr(network *net.IPNet, divisor int64) ([]net.IPNet, error) {
```

Comment:
Do you think some of this logic should go to Core? Most logic is now in Core to perform such operations.